index.d.ts

```diff
@@ -0,0 +1,398 @@
+	 *
+	 * @default true
+	 */
+	readonly preferLocal?: boolean;
+
+	/**
+	 * Preferred path to find locally installed binaries in (use with `preferLocal`).
+	 *
+	 * @default process.cwd()
+	 */
+	readonly localDir?: string;
+
+	/**
+	 * Setting this to `false` resolves the promise with the error instead of rejecting it.
+	 *
+	 * @default true
+	 */
+	readonly reject?: boolean;
+
+	/**
+	 * Keep track of the spawned process and `kill` it when the parent process exits.
+	 *
+	 * @default true
+	 */
+	readonly cleanup?: boolean;
+
+	/**
+	 * Specify the character encoding used to decode the `stdout` and `stderr` output. If set to `null`, then `stdout` and `stderr` will be a `Buffer` instead of a string.
+	 *
+	 * @default 'utf8'
+	 */
+	readonly encoding?: EncodingType;
+
+	/**
+	 * If `timeout` is greater than `0`, the parent will send the signal identified by the `killSignal` property (the default is `SIGTERM`) if the child runs longer than `timeout` milliseconds.
+	 *
+	 * @default 0
+	 */
+	readonly timeout?: number;
+
+	/**
+	 * Buffer the output from the spawned process. When buffering is disabled you must consume the output of the `stdout` and `stderr` streams because the promise will not be resolved/rejected until they have completed.
+	 *
+	 * @default true
+	 */
+	readonly buffer?: boolean;
+
+	/**
+	 * Largest amount of data in bytes allowed on `stdout` or `stderr`. Default: 10MB.
+	 *
+	 * @default 10000000
+	 */
+	readonly maxBuffer?: number;
+
+	/**
+	 * Signal value to be used when the spawned process will be killed.
+	 *
+	 * @default 'SIGTERM'
+	 */
+	readonly killSignal?: string | number;
+
+	/**
+	 * Same options as [`stdio`](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_options_stdio).
+	 *
+	 * @default 'pipe'
+	 */
+	readonly stdin?: StdIOOption;
+
+	/**
+	 * Same options as [`stdio`](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_options_stdio).
+	 *
+	 * @default 'pipe'
+	 */
+	readonly stdout?: StdIOOption;
+
+	/**
+	 * Same options as [`stdio`](https://nodejs.org/dist/latest-v6.x/docs/api/child_process.html#child_process_options_stdio).
+	 *
+	 * @default 'pipe'
+	 */
+	readonly stderr?: StdIOOption;
+
+	/**
+	 * If `true`, no quoting or escaping of arguments is done on Windows. Ignored on other platforms. This is set to `true` automatically when the `shell` option is `true`.
+	 *
+	 * @default false
+	 */
+	readonly windowsVerbatimArguments?: boolean;
+}
+
+export interface Options<EncodingType = string>
+	extends CommonOptions<EncodingType> {
+	/**
+	 * Write some input to the `stdin` of your binary.
+	 */
+	readonly input?: string | Buffer | ReadableStream;
+}
+
+export interface SyncOptions<EncodingType = string>
+	extends CommonOptions<EncodingType> {
+	/**
+	 * Write some input to the `stdin` of your binary.
+	 */
+	readonly input?: string | Buffer;
+}
+
+export interface ExecaReturnBase<StdOutErrType> {
```

Comment:
```suggestion
export interface ExecaReturnBase<StdoutErrorType> {
```

Reply:
Hmm, actually, it's just just the `stdout` type, but also `stderr`. The naming might make it confusing for users.